### ISSUE TITLE
secondary annotation layer always visible / visibility cannot be controlled

### ISSUE THREAD

**Issue by celmi** [qgis/QGIS#48584]
### What is the bug or the crash?

This blog here
https://north-road.com/2021/10/21/introducing-annotation-layers-in-qgis-3-22/
describes, how you can add custom / secondary annotation layers that appear in the list of layers and whose visibility can be controlled with the checkbox.
However, if I check/uncheck the checkbox next to a custom annotation layer, nothing happens, Annotation of that particular layer is always visible.

### Steps to reproduce the issue

create secondary annotation layer
activate layer, create text annotation
deactivate visibility checkbox in list of layer

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.22.6-Białowieża | QGIS code revision | b6f33b1a
-- | -- | -- | --
Qt version | 5.15.2
Python version | 3.9.5
GDAL/OGR version | 3.4.2
PROJ version | 9.0.0
EPSG Registry database version | v10.041 (2021-12-03)
GEOS version | 3.10.2-CAPI-1.16.0
SQLite version | 3.38.1
PDAL version | 2.3.0
PostgreSQL client version | 13.0
SpatiaLite version | 5.0.1
QWT version | 6.1.3
QScintilla2 version | 2.11.5
OS version | Windows 10 Version 2009
  |   |   |  
Active Python plugins
swiss_locator | 1.4.4
db_manager | 0.1.20
grassprovider | 2.12.99
MetaSearch | 0.3.5
processing | 2.12.99
sagaprovider | 2.12.99

</body></html>QGIS version
3.22.6-Białowieża
QGIS code revision
[b6f33b1a](https://github.com/qgis/QGIS/commit/b6f33b1a)
Qt version
5.15.2
Python version
3.9.5
GDAL/OGR version
3.4.2
PROJ version
9.0.0
EPSG Registry database version
v10.041 (2021-12-03)
GEOS version
3.10.2-CAPI-1.16.0
SQLite version
3.38.1
PDAL version
2.3.0
PostgreSQL client version
13.0
SpatiaLite version
5.0.1
QWT version
6.1.3
QScintilla2 version
2.11.5
OS version
Windows 10 Version 2009

Active Python plugins
swiss_locator
1.4.4
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.5
processing
2.12.99
sagaprovider
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by celmi** [comment]
Update: I think only Textboxes are affected by this bug. I changed some wordings

**Comment by nyalldawson** [comment]
Which "textbox" annotation are you referring to? If it's the one with a "speech bubble" type background, then this is expected -- that's not actually an annotation item content.

**Comment by celmi** [comment]
Yes, that's the one. Upon hover I get the mouse-over information that it's called "Text annotations". The dropdown options offers variants called "Form annotation", "SVG annotation", "HTML annotation" and "move annotation". 
It's actually part of the "annotations toolbar". That's why I would have expected that such elements will be drawn to the annotation layer. I'm finding it very confusing, that some elements of the annotations toolbar will be drawn to that layer, and others not.

**Comment by celmi** [comment]
Can someone from the dev-team please decide, if this is a bug or a feature? IMHO it is a either a bug or a feature with ill UI-design. In the latter case I would propose to convert the issue to a feature request (enhance user-guidance, so the different layer-behaviour of different annotation-tools is intuitively clear to users).

**Comment by celmi** [comment]
Just a nudge to prevent the evil robot from closing this issue automatically...

**Comment by celmi** [comment]
nudge

**Comment by YoannQDQ** [comment]
@celmi Why not create a new annotation layer, then link your speech-bubble-annotation to it? Thus, when you toggle the annotation layer in the LayerTreeView, every linked "speech-bubble-annotation" is also toggled.


![link_annotations](https://user-images.githubusercontent.com/9693475/233868511-b6159b91-9f4f-4ebc-bf37-8fa209b1fa39.png)

I aggree that it can be confusing to have two very different kind of annotations though.
